### PR TITLE
cinder: Add support for the KVM virtio-scsi driver

### PR DESCRIPTION
**Pull request by mikebryant**
**What this PR does / why we need it**:

The VirtIO SCSI driver for KVM changes the way disks appear in /dev/disk/by-id.
This adds support for the new format.
Without this, volume attaching on an openstack cluster using this kvm driver doesn't work

**Special notes for your reviewer**:
Does this need e2e tests? I couldn't find anywhere to add another openstack configuration used in the e2e tests.

Wiki page about this: https://wiki.openstack.org/wiki/Virtio-scsi-for-bdm

**Release note**:

```release-note
cinder: Add support for the KVM virtio-scsi driver
```

### CODE REVIEW

**Comment by idvoretskyi** [comment]
cc @k8s-sig-openstack-pr-reviews

**Review by bradbeam — APPROVED** [review]

**Comment by mikebryant** [comment]
Is there anything I need to do here?

**Comment by xsgordon** [comment]
@k8s-bot ok to test

**Review by hogepodge — APPROVED** [review]
lgtm

Currently, afaict, there are no dependent e2e tests that would need to be updated. This patch should be considered a bug fix, as OpenStack can provide Cinder volumes as scsi devices.

**Comment by mikebryant** [comment]
@hogepodge Should your comment have applied the lgtm label?

**Comment by hogepodge** [comment]
@mikebryant I don't have approver status, afaik.

**Comment by stuart-warren** [comment]
ping @anguslees

**Comment by xsgordon** [comment]
@idvoretskyi thoughts? To me this is release-note-none worthy.

**Comment by anguslees** [comment]
Sorry for the delay, I am no longer working on the openstack cloud provider and no longer have access to openstack clusters for testing :(

This lgtm to me though:
/approve

**Comment by mikebryant** [comment]
@anguslees Thanks for taking the time :)
Sorry for bugging you, you're currently listed as the only owner of the openstack cloudprovider: https://github.com/kubernetes/kubernetes/blob/2c6fbc95c43b9f8de41903361a9a286dba417f23/pkg/cloudprovider/providers/openstack/OWNERS



Now... who do I need to ping to get an lgtm?

**Comment by anguslees** [comment]
@mikebryant: Yes I know, this is why I haven't removed my name from there yet - I figured one disinterested and occasional reviewer is still better than no reviewers :(

Huh, I thought /approve was sufficient, let's try:
/lgtm

**Comment by anguslees** [comment]
@k8s-bot cvm gce e2e test this

**Comment by mikebryant** [comment]
@anguslees Thanks muchly! :)

**Pull request merged**
